### ISSUE TITLE
Feature request: different backup options

### ISSUE THREAD

**Issue by inariksit** [GrammaticalFramework/gf-ud#22]
Two things: Backup cat + funs, and string literal funs.

### Backup* funs

In the current master branch, Backup* funs are added by default, if some parts of the sentence can't be included in the GF tree. However, this requires that the grammar contains such funs, which not all grammars do. So I would like to have this feature optional.

It can either be a command line argument, or the Haskell code can check whether the PGF contains a cat called Backup and funs called BackupNP etc., and use them only if they are found in the PGF.

I don't care about which way to use, just to have the feature optional in some way.

### String literals for OOV words

If the sentence contains words that are not in the lexicon, I would like to create those words as string literals. So "mimsy were the borogroves" would result in an otherwise normal GF tree, but with the subtrees `StrA "mimsy"` and `StrN "borogrove"`.

Again, this feature should be optional: either command line arg, or check if the grammar contains `StrA : String -> A`.

### Future work: modify the PGF grammar?

The new majestic runtime will allow modifying PGFs on the fly. So when that is possible, I'd prefer to create proper lexicon entries `mimsy_A` and `borogrove_N`, using real GF smart paradigms, and insert them into the PGF. Similarly, the Backup* funs would also be possible to insert into the PGF.

So once the new runtime is in place, I think that command line argument would be a better option. And if these features are added into gf-ud already before majestic runtime, it makes sense to just use command line arguments from the beginning. (I don't care which option is the default.)
